superset/utils.py

```diff
@@ -877,8 +877,15 @@ def split_adhoc_filters_into_base_filters(fd):
                     sql_where_filters.append(adhoc_filter.get('sqlExpression'))
                 elif clause == 'HAVING':
                     sql_having_filters.append(adhoc_filter.get('sqlExpression'))
-        fd['where'] = ' AND '.join(['({})'.format(sql) for sql in sql_where_filters])
-        fd['having'] = ' AND '.join(['({})'.format(sql) for sql in sql_having_filters])
-        fd['having_filters'] = simple_having_filters
-        fd['filters'] = simple_where_filters
-        del fd['adhoc_filters']
```

Comment:
Note we can no longer remove the `adhoc_filters` field as we need to ensure that ad-hoc filters persist. Note currently by deleting this field the ad-hoc filters actually get rebuilt on the frontend by merging the legacy and ad-hoc filters.